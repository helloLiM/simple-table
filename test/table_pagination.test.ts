import { shallowMount,mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import tablePagination from "../src/components/table_pagination";
import { nextTick } from 'vue';
import * as _ from 'lodash';

const data = [
  {
      id: '0',
      name: 'a',
      sex: 'male',
      age: 12,
  }, {
      id: '1',
      name: 'b',
      sex: 'female',
      age: 64,
  }];

// 点击上一页/下一页
test('test click page', async () => {
    const wrapper = shallowMount(tablePagination, {
        props: {
            allData : data,
            pageLimit : 1
          },
    })
  const clickPrev = wrapper.find('.table-pagination_prev')
  const clickNext = wrapper.find('.table-pagination_next')
//   在第一页点击上一页
  clickPrev.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 1 页')).toBe(true)
//   点击下一页
  clickNext.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 2 页')).toBe(true)
// 在最后一页点下一页
  clickNext.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 2 页')).toBe(true)
  // 在第二页点击 上一页
  clickPrev.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 1 页')).toBe(true)

})

// 点击跳页
test('test click page', async () => {
  const wrapper = shallowMount(tablePagination, {
      props: {
          allData : data,
          pageLimit : 1,
          pageJump : 2,
      },
  })
  const clickJump = wrapper.find('.table-pagiantion_jump')
  clickJump.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 2 页')).toBe(true)
})

// 页码大于总页数时点击跳页
test('test click page', async () => {
  const wrapper = shallowMount(tablePagination, {
      props: {
          allData : data,
          pageLimit : 1,
          pageJump : 5.3,
      },
  })
  const clickJump = wrapper.find('.table-pagiantion_jump')
  clickJump.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 2 页')).toBe(true)
})

// 页码小于总页数时点击跳页
test('test click page', async () => {
  const wrapper = shallowMount(tablePagination, {
      props: {
          allData : data,
          pageLimit : 1,
          pageJump : -5,
      },
  })
  const clickJump = wrapper.find('.table-pagiantion_jump')
  clickJump.trigger('click')
  await nextTick()
  expect(wrapper.html().includes('第 1 页')).toBe(true)
})
