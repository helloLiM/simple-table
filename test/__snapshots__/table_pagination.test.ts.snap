// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"table-pagination\\">
  <ul>
    <li class=\\"table-pagination_prev\\">上一页</li>
    <li class=\\"table-pagination_next\\">下一页</li>
    <li class=\\"table-pagiantion_info\\">共 33 条</li>
    <li class=\\"table-pagiantion_info\\">第 1 页</li>
    <li class=\\"table-pagiantion_info\\">共 4 页</li>
    <li class=\\"table-pagiantion_info\\"><input class=\\"inputPage\\" type=\\"number\\"><span class=\\"table-pagiantion_jump\\">跳转</span></li>
  </ul>
</div>"
`;
